spyder/utils/introspection/utils.py

```diff
@@ -210,6 +210,45 @@ def get_keywords(lexer):
                 continue
     return keywords
 
+def get_words_file(file_path=None, content=None, extension=None):
+    """
+    Extract all words from a source code file to be used in code completion.
+
+    Extract the list of words that contains the file in the editor,
+    to carry out the inline completion similar to VSCode.
+    """
+    if file_path == None and content != None and extension != None:
```

Comment:
@malliwi88 I suggest we do something like this:

```python
if file_path and content is None and extension is None:
     extension = os.path.splitext(file_path)[1]

Then the logic for the rest of the function is the same and not repeated.
So that the logic of the method is easier to follow